### ISSUE TITLE
Separate usage of the Server and the ServerConnector Executors

### ISSUE THREAD

**Issue by sbordet** [eclipse/jetty.project#521]
It is currently possible to specify one `Executor` for the `Server` and a different `Executor` for the `ServerConnector`.

This is useful for people that would like to prioritize NIO tasks run by the `ServerConnector` `Executor` and leave other tasks at a lower priority.

Unfortunately, Jetty is not rigorous in using the right `Executor` when running non-NIO tasks.
For example, `HttpChannel` uses the `ServerConnector` `Executor` rather than the `Server` `Executor`.

A code review is needed to separate usage of `Executor`s properly, so that only NIO tasks use the `ServerConnector` `Executor`, while all the rest uses the `Server` `Executor`.

**Comment by gregw** [comment]
I've fixed up the HttpChannel usage and had a little look around for others.  Nothing else found yet, but will leave open so we can have a more thorough check.

**Comment by sbordet** [comment]
@gregw have you looked at the client(s) too ?

**Comment by gregw** [comment]
Nope.

**Comment by gregw** [comment]
I had a look at this for the client and it doesn't really make much sense there as there is no server & connector that may have different thread pools, there is just the client.   It would need the client API to be changed to support different pools for different purposes.

Note also that with EWYK scheduling, threads may cross boundaries any way, so a thread from a producers threadpool is likely to end up consuming as well, even if the consumer has it's own thread pool.   

Assigning to you for your consideration.

**Comment by dain** [comment]
What version will this be in?

**Comment by gregw** [comment]
It will be in the next 9.3.9 release.

Note that in discussion with Simone, we are not sure that it is not actually worthwhile having separate threadpools for the connector and server.

Firstly if we are using EWYK strategy, then a thread that has done a connector task to produce another task will immediately consume that task (with a hot cache), so a thread from the connector pool can consume jobs that are nominally for the server pool.

But more importantly, I think it is generally a bad idea.  If you have high and low priority areas of the code, then it is going to be far better to simply change the priority of the current thread rather than arrange to dispatch to a different thread of different priority, which will incur dispatch delays and a cold cache.

In the low level scheduling, we already give priority to some selected tasks (ie we directly run tasks that will unblock a writer rather than dispatch them).    We are now investigating giving similar priority to some tasks in HTTP/2 (eg processing window updates) so that they do not become stalled behind execution queues.

So in summary, the only reason I can see to have different thread pools is to have different execution queues.  If the difference is anything to do with the thread itself (eg priority), then I don't think the dual pool approach is valid.

cheers
